### PR TITLE
Release v0.6

### PR DESCRIPTION
**Pull request by jiangliu**
Release vm-memory v0.6, with latest dirty track bitmap implementation.

### CODE REVIEW

**Review by sboeuf — DISMISSED** [review]

**Comment by andreeaflorescu** [comment]
@jiangliu can we push this a bit until we fix this comment: https://github.com/rust-vmm/vm-memory/pull/125#discussion_r643697285

@lauralt wants to submit a PR with it.

**Comment by jiangliu** [comment]
> @jiangliu can we push this a bit until we fix this comment: [#125 (comment)](https://github.com/rust-vmm/vm-memory/pull/125#discussion_r643697285)
> 
> @lauralt wants to submit a PR with it.

sure, then I will try to create a patch for #149

**Comment by bonzini** [comment]
I wonder if it's a good idea to break the exported types due to the dirty bitmap implementation. Perhaps export the new version from some sub-package, and leave those exported directly from `vm_memory` as aliases with `()` as the bitmap type?

**Comment by jiangliu** [comment]
> I wonder if it's a good idea to break the exported types due to the dirty bitmap implementation. Perhaps export the new version from some sub-package, and leave those exported directly from `vm_memory` as aliases with `()` as the bitmap type?

I have give a try, but it always breaks existing users of GuestMemoryRegion. For example, I have tried to rename "GuestMemoryRegion" to "GuestMemoryRegionImpl", and defines
```
pub trait GuestMemoryRegion: GuestMemoryRegionImpl<B = ()> {}
```
Then existing users need to "use GuestMemoryRegionImpl" too, which still breaks:(

**Comment by jiangliu** [comment]
I have tried several times to avoid breaking compatibility but failed:(
So should we release v1.0.0 instead of v0.6.0?

**Comment by andreeaflorescu** [comment]
We don't really give any guarantees related to the interface until we reach 1.0.0. I think we were also considering some other interface simplifications (#141), and I would be more inclined to first agree on those, implement them, and only then release 1.0.0.

For this one, I would be fine with 0.6.0 even if we break the interface.

File: src/volatile_memory.rs [hunk 1]
```diff
@@ -1574,6 +1576,7 @@ mod tests {
         // somehow and this helps to ensure the sleep actually happens before the store rather then
```
Comment (lauralt) [inline src/volatile_memory.rs:1576]: I would update the comments, so they do not mention about `sleep` anymore.

Comment (jiangliu) [reply src/volatile_memory.rs]: fixed

**Review by alexandruag — APPROVED** [review]

**Pull request merged**